x/programs/examples/simulator/cmd/program.go

```diff
@@ -0,0 +1,307 @@
+			}
+			var err error
+			programID, err = deployProgram(context.Background(), &action)
+			if err != nil {
+				return err
+			}
+
+			utils.Outf("{{green}}deploy transaction successful: {{/}} %v\n\n", programID.String())
+		case "call":
+			utils.Outf("{{yellow}}max fee:{{/}} %v\n", action.MaxFee)
+			resp, id, err := callProgram(context.Background(), &action, &p.Config)
+			if err != nil {
+				return err
+			}
+			utils.Outf("{{green}}call transaction successful: {{/}} %s\n", id.String())
+			utils.Outf("{{blue}}response: {{/}}%d\n\n", resp)
+		default:
+			return fmt.Errorf("%w: %s", ErrInvalidAction, action.Name)
+		}
+	}
+
+	return nil
+}
+
+func deployProgram(ctx context.Context, action *Action) (ids.ID, error) {
+	programBytes, err := os.ReadFile(action.ProgramPath)
+	if err != nil {
+		return ids.Empty, err
+	}
+	// simulate create program transaction
+	programID, err = generateRandomID()
+	if err != nil {
+		return ids.Empty, err
+	}
+	// store the program to disk
+	err = storage.SetProgram(ctx, db, programID, programBytes)
+	if err != nil {
+		return ids.Empty, err
+	}
+
+	return programID, nil
+}
+
+func callProgram(ctx context.Context, action *Action, config *Config) (uint64, ids.ID, error) {
+	// get program ID from deploy action if set to inherit
+	var programIDBytes = make([]byte, 32)
+	if action.ProgramID == inheritIDKey {
+		copy(programIDBytes, programID[:])
+	} else {
+		copy(programIDBytes, []byte(action.ProgramID))
+	}
+
+	programID, err := ids.ToID(programIDBytes)
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+
+	// get program bytes from disk
+	programBytes, ok, err := storage.GetProgram(ctx, db, programID)
+	if !ok {
+		return 0, ids.Empty, fmt.Errorf("%w: %s", ErrProgramNotFound, programID)
+	}
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+
+	// simulate call program transaction
+	callID, err := generateRandomID()
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+
+	cfg, err := newConfig(action, config)
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+
+	// TODO: handle custom imports
+	supported := runtime.NewSupportedImports()
+	supported.Register("state", func() runtime.Import {
+		return pstate.New(log, db)
+	})
+	supported.Register("program", func() runtime.Import {
+		return program.New(log, db)
+	})
+
+	rt := runtime.New(log, cfg, supported.Imports())
+	defer rt.Stop()
+	err = rt.Initialize(ctx, programBytes)
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+
+	// get function params
+	params, err := createParams(ctx, programID, rt.Memory(), db, action.Parameters)
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+
+	resp, err := rt.Call(ctx, action.Function, params...)
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+
+	// only commit to state if the call is successful
+	err = db.Commit(ctx)
+	if err != nil {
+		return 0, ids.Empty, err
+	}
+	utils.Outf("{{yellow}}fee balance: {{/}}%d\n", rt.Meter().GetBalance())
+	return resp[0], callID, nil
+}
+
+func generateRandomID() (ids.ID, error) {
```

Comment:
You can do `ids.GenerateTestID`

Reply:
the issue with that is its not random. so if we call the simulator again it will generate the same ids.

https://github.com/ava-labs/avalanchego-internal/blob/f79d6099d3514c31f139be9a97ef543645ecc361/ids/test_generator.go#L8

Reply:
added